Tasks/Common/azure-arm-rest-v2/AzureServiceClientBase.ts

```diff
@@ -140,32 +140,46 @@ export class AzureServiceClientBase {
         var waitIndefinitely = timeoutInMinutes == 0;
+        var ignoreTimeoutErrorThreshold = 5;
         var request = new webClient.WebRequest();
         request.method = "GET";
         request.uri = response.headers["azure-asyncoperation"] || response.headers["location"];
         if (!request.uri) {
             throw new Error(tl.loc("InvalidResponseLongRunningOperation"));
         }
         while (true) {
-            response = await this.beginRequest(request);
-            tl.debug(`Response status code : ${response.statusCode}`);
-            if (response.statusCode === 202 || (response.body && (response.body.status == "Accepted" || response.body.status == "Running" || response.body.status == "InProgress"))) {
-                if(response.body && response.body.status) {
-                    tl.debug(`Response status : ${response.body.status}`);
+            try {
+                response = await this.beginRequest(request);
+                tl.debug(`Response status code : ${response.statusCode}`);
+                if (response.statusCode === 202 || (response.body && (response.body.status == "Accepted" || response.body.status == "Running" || response.body.status == "InProgress"))) {
+                    if (response.body && response.body.status) {
+                        tl.debug(`Response status : ${response.body.status}`);
+                    }
+                    // If timeout; throw;
+                    if (!waitIndefinitely && timeout < new Date().getTime()) {
+                        throw new Error(tl.loc("TimeoutWhileWaiting"));
+                    }
+
+                    // Retry after given interval.
+                    var sleepDuration = 15;
+                    if (response.headers["retry-after"]) {
+                        sleepDuration = parseInt(response.headers["retry-after"]);
+                    }
+                    await this.sleepFor(sleepDuration);
+                } else {
+                    break;
                 }
-                // If timeout; throw;
-                if (!waitIndefinitely && timeout < new Date().getTime()) {
-                    throw new Error(tl.loc("TimeoutWhileWaiting"));
+            }
+            catch (error) {
+                let errorString: string = (error && error.toString()) || "";
+                if(errorString && errorString.toLowerCase().indexOf("request timeout") >= 0 && ignoreTimeoutErrorThreshold > 0) {
+                    // Ignore Request Timeout error and continue polling operation
+                    tl.debug(`Request Timeout: ${request.uri}`);
+                    ignoreTimeoutErrorThreshold--;
                 }
-
-                // Retry after given interval.
-                var sleepDuration = 15;
-                if (response.headers["retry-after"]) {
-                    sleepDuration = parseInt(response.headers["retry-after"]);
+                else {
+                    throw error;
```

Comment:
just `throw`. Not sure about javascript but in C# `throw error` would change the call stack

Reply:
In javascript, you can re throw the error which will preserve the stack by doing: `throw error`. `throw new Error(error)` will lead to stack trace being lost.